### ISSUE TITLE
pulse rms different between python and cython

### ISSUE THREAD

**Issue by joefowler** [usnistgov/mass#39]
**[Original report](https://bitbucket.org/joe_fowler/mass/issues/39) by Dan Becker (Bitbucket: [danbek](https://bitbucket.org/danbek), GitHub: [danbek](https://github.com/danbek)).**

----------------------------------------

Cython uses:

```
#!python
pulse_avg = pulse_sum / (nSamples - nPresamples - 2) - ptm
p_pulse_average_array[j] = <float>pulse_avg
p_pulse_rms_array[j] = <float>sqrt(pulse_rms_sum / (nSamples - nPresamples - 2)
                                   - ptm*pulse_avg*2 + ptm**2)
```
python uses:
```
#!python
self.p_pulse_average[first:end] = self.data[:seg_size, self.nPresamples:].mean(axis=1)
ptm = self.p_pretrig_mean[first:end]
self.p_pulse_average[first:end] -= ptm
self.p_pulse_rms[first:end] = np.sqrt(
    (self.data[:seg_size, self.nPresamples:]**2.0).mean(axis=1) -
    ptm*(ptm + 2*self.p_pulse_average[first:end]))
```

Thse are not the same. I believe that the python code is correct, but under the cython approach, the last line should be:

```
#!python
p_pulse_rms_array[j] = <float>sqrt(pulse_rms_sum / (nSamples - nPresamples - 2)
                                   - 2*ptm*pulse_avg - ptm**2)
```

(I also moved the multiplication by 2 to make it more clear).

This code is confusing because of the subtraction of the pretrigger mean from the pulse average (and this is perhaps why this bug was introduced). I think a comment would be useful.

**Comment by joefowler** [comment]
**Original comment by Young Il Joe (Bitbucket: [jojo978](https://bitbucket.org/jojo978), GitHub: [jojo978](https://github.com/jojo978)).**

----------------------------------------

Bug fixes for pulse_rms calculations. fixes #39,

**Comment by joefowler** [comment]
**Original comment by Young Il Joe (Bitbucket: [jojo978](https://bitbucket.org/jojo978), GitHub: [jojo978](https://github.com/jojo978)).**

----------------------------------------

Thanks Dan, I just fixed this one. But I kept ptm*pulse_avg*2 just because I'm not sure how type coercion works in C.
